app/server/middleware/repos.js

```diff
@@ -28,34 +19,37 @@ module.exports = {
     listReposPermission (req, res, next) {
         console.log('looking up user permission for repos');
 
-        let user = req.query.permission_user,
+        let username = req.query.permission_user,
             repos = req.entity;
 
-        if (user) {
-            repos.forEach((repo) => {
-                repo.permission = 'read';
+        if (username) {
+            repoRepository.getPermissions(username, repos).then((repos) => {
+                req.entity = repos;
+                next();
+            }).catch((err) => {
+                next(err);
             });
+        } else {
+            next();
         }
-        next();
     },
 
     listReposLinks (req, res, next) {
         console.log('checking for links on repo list');
 
-        let user = req.query.permission_user,
+        let username = req.query.permission_user,
             repos = req.entity;
 
-        if (user) {
+        if (username) {
             repos.forEach((repo) => {
-                repo.links = [{
-                    rel: 'edit-user-permission',
-                    href: 'repos/' + repo.id + '/users/' + user + '/permissions/{permission}',
-                    method: 'PUT'
-                }, {
-                    rel: 'remove-user-permission',
-                    href: 'repos/' + repo.id + '/users/' + user,
-                    method: 'DELETE'
-                }];
+                let permission = repo.permission;
+                if (permission.permissive === 'admin' || permission.github === 'admin') {
```

Comment:
I'm not sure this is the correct logic. I think this will limit the edit links to only those repos that the *selected* user has admin on, rather than the *logged in* user. We might need to file a separate ticket for that, as it's probably much more complexity than what's in place now.